egui/src/widgets/plot/mod.rs

```diff
@@ -216,6 +221,30 @@ impl Plot {
         self
     }
 
+    /// Provide a function to customize the labels for the X axis.
+    ///
+    /// This is useful for custom input domains, e.g. date/time.
+    ///
+    /// If axis labels should not appear for certain values or beyond a certain zoom/resolution,
+    /// the formatter function can return empty strings. This is also useful if your domain is
+    /// discrete (e.g. only full days in a calendar).
+    pub fn x_axis_formatter<F: 'static + Fn(f64) -> String>(mut self, func: F) -> Self {
```

Comment:
nit: I prefer this (and it is more consistent with the rest of egui)

```suggestion
    pub fn x_axis_formatter(mut self, func: impl Fn(f64) -> String + 'static) -> Self {
```